From 93e20bade9f38303ac33b2245320de4b65c66bb0 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:36:30 -0400
Subject: [ethernet] i40e/i40evf: Some flow director HW definition fixes

Message-id: <1409658042-22689-30-git-send-email-sassmann@redhat.com>
Patchwork-id: 90224
O-Subject: [RHEL7.1 PATCH 029/281] i40e/i40evf: Some flow director HW definition fixes
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Anjali Singhai Jain <anjali.singhai@intel.com>

1) Fix a name of the error bit to correctly indicate the error.
2) Added a fd_id field in the 32 byte desc at the place(qw0) where it gets
reported in the programming error desc WB. In a normal data desc
the fd_id field is reported in qw3.

Change-ID: Ide9a24bff7273da5889c36635d629bc3b5212010
Signed-off-by: Anjali Singhai Jain <anjali.singhai@intel.com>
Acked-by: Shannon Nelson <shannon.nelson@intel.com>
Signed-off-by: Kevin Scott <kevin.c.scott@intel.com>
Signed-off-by: Catherine Sullivan <catherine.sullivan@intel.com>
Tested-by: Kavindya Deegala <kavindya.s.deegala@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 77e29bc6fc814a2283c9dda07d24f2efb53d585c)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_type.h b/drivers/net/ethernet/intel/i40e/i40e_type.h
index 181a825..5c902f4 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_type.h
+++ b/drivers/net/ethernet/intel/i40e/i40e_type.h
@@ -458,6 +458,10 @@ union i40e_32byte_rx_desc {
    union {
     __le32 rss; /* RSS Hash */
     __le32 fcoe_param; /* FCoE DDP Context id */
+    /* Flow director filter id in case of
+     * Programming status desc WB
+     */
+    __le32 fd_id;
    } hi_dword;
   } qword0;
   struct {
@@ -698,7 +702,7 @@ enum i40e_rx_prog_status_desc_prog_id_masks {
 enum i40e_rx_prog_status_desc_error_bits {
  /* Note: These are predefined bit offsets */
  I40E_RX_PROG_STATUS_DESC_FD_TBL_FULL_SHIFT = 0,
- I40E_RX_PROG_STATUS_DESC_NO_FD_QUOTA_SHIFT = 1,
+ I40E_RX_PROG_STATUS_DESC_NO_FD_ENTRY_SHIFT = 1,
  I40E_RX_PROG_STATUS_DESC_FCOE_TBL_FULL_SHIFT = 2,
  I40E_RX_PROG_STATUS_DESC_FCOE_CONFLICT_SHIFT = 3
 };
-- 
1.7.1